### PR TITLE
feat(cts): add system tracker support

### PR DESCRIPTION
**Pull request by ShiChangkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add system tracker support
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST='./huaweicloud/services/acceptance/cts' TESTARGS='-run TestAccCTSTracker_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cts -v -run TestAccCTSTracker_basic -timeout 360m -parallel 4
=== RUN   TestAccCTSTracker_basic
=== PAUSE TestAccCTSTracker_basic
=== CONT  TestAccCTSTracker_basic
--- PASS: TestAccCTSTracker_basic (92.81s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cts       92.955s
```

### CODE REVIEW

File: docs/resources/cts_tracker.md [hunk 1]
```diff
@@ -1,76 +1,64 @@
...
+* `validate_file` - (Optional, Bool) Specifies whether trace file verification is enabled during trace transfer.
 
-* `operations` - (Required, String) Trigger conditions for sending a notification.
+* `kms_id` - (Optional, String) Specifies the ID of KMS key used for trace file encryption.
 
-* `is_send_all_key_operation` - (Required, Bool) When the value is **false**, operations cannot be left empty.
-
-* `need_notify_user_list` - (Optional, String) The users using the login function. When these users log in,
-  notifications will be sent.
+* `status` - (Optional, String) Specifies the tracker status, the value can be **enabled** or **disabled**.
```
Comment (Jason-Zhang9309) [inline docs/resources/cts_tracker.md]: How about use bool type parameter such as `enable`?

File: huaweicloud/services/cts/resource_huaweicloud_cts_tracker.go [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+			},
+
+			"bucket_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"file_prefix": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"bucket_name"},
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/cts/resource_huaweicloud_cts_tracker.go:51]: suggest to add a validatefunc

File: huaweicloud/services/acceptance/cts/resource_huaweicloud_cts_tracker_test.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
+package deprecated
```
Comment (niuzhenguo) [inline huaweicloud/services/acceptance/cts/resource_huaweicloud_cts_tracker_test.go]: s/deprecated/cts

File: huaweicloud/services/acceptance/cts/resource_huaweicloud_cts_tracker_test.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+func testAccCheckCTSTrackerExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmtp.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmtp.Errorf("No ID is set")
```
Comment (niuzhenguo) [inline huaweicloud/services/acceptance/cts/resource_huaweicloud_cts_tracker_test.go]: `fmtp` and `fmt` used mixed in test file.

File: huaweicloud/services/cts/resource_huaweicloud_cts_tracker.go [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+			},
+			"lts_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"validate_file": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"kms_id": {
```
Comment (niuzhenguo) [inline huaweicloud/services/cts/resource_huaweicloud_cts_tracker.go]: should we need to add `RequiredWith` bucket_name for `kms_id` and `validte_file` as well?

File: docs/resources/cts_tracker.md [hunk 6]
```diff
@@ -1,76 +1,64 @@
...
 
-* `create` - Default is 10 minute.
-* `delete` - Default is 10 minute.
+* `create` - Default is 5 minute.
+* `delete` - Default is 5 minute.
 
 ## Import
 
-CTS tracker can be imported using  `tracker_name`, e.g.
+CTS tracker can be imported using `name`. Currently, only **system** is available. e.g.
```
Comment (Lance52259) [inline docs/resources/cts_tracker.md:62]: If tracker only supports **system** type and import by type, not ID, I think we should call this resource `huaweicloud_cts_systracker`

File: huaweicloud/services/acceptance/cts/resource_huaweicloud_cts_tracker_test.go [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+		if name == "" {
+			return "", fmt.Errorf("CTS tracker resource %s not found", tracker.Primary.ID)
+		}
+		return name, nil
+	}
+}
+
+func testAccCheckCTSTrackerDestroy(s *terraform.State) error {
+	return nil
+}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/cts/resource_huaweicloud_cts_tracker_test.go:72]: destroy should test whether the tracker state is `disabled`.

File: docs/resources/cts_tracker.md (hunk 6 above)
Comment (ShiChangkuo) [reply docs/resources/cts_tracker.md:62]: will support type **data** tracker later

File: huaweicloud/services/acceptance/cts/resource_huaweicloud_cts_tracker_test.go (hunk 7 above)
Comment (ShiChangkuo) [reply huaweicloud/services/acceptance/cts/resource_huaweicloud_cts_tracker_test.go:72]: when the system tracker is destroyed, the status just keep it as is

File: docs/resources/cts_tracker.md (hunk 1 above)
Comment (ShiChangkuo) [reply docs/resources/cts_tracker.md]: sure, fixed

File: huaweicloud/services/cts/resource_huaweicloud_cts_tracker.go (hunk 2 above)
Comment (ShiChangkuo) [reply huaweicloud/services/cts/resource_huaweicloud_cts_tracker.go:51]: fixed

File: huaweicloud/services/acceptance/cts/resource_huaweicloud_cts_tracker_test.go (hunk 3 above)
Comment (ShiChangkuo) [reply huaweicloud/services/acceptance/cts/resource_huaweicloud_cts_tracker_test.go]: fixed

File: huaweicloud/services/cts/resource_huaweicloud_cts_tracker.go (hunk 5 above)
Comment (ShiChangkuo) [reply huaweicloud/services/cts/resource_huaweicloud_cts_tracker.go]: fixed

**Pull request merged**